hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azure/NativeAzureFileSystem.java

```diff
@@ -1242,7 +1242,10 @@ public void setEncodedKey(String anEncodedKey) {
      * when the stream is closed.
      */
     private void restoreKey() throws IOException {
-      store.rename(getEncodedKey(), getKey());
+      String key = getKey();
+      FileMetadata existingMetadata = store.retrieveMetadata(key);
```

Comment:
Lets not call this call as discussed in the other comment around how to get etag.
+
By the emptySrc file was created, till this moment some other process would have made change to that path and hence etag would change. so, now if we read etag here, we will get the wrong etag.

Reply:
We want the latest eTag on the path, not the initial eTag. Hence we need to fetch the latest eTag on that path. Simply passing the initial eTag won't help here

Reply:
As I understand from the code, there shouldn't be any write on the original file and on the close of outputstream on the tmpFile, it rename it to original file.
Is there any flow which would be writing on the original file before the rename.